### PR TITLE
Pass correct process identifier to auxiliary processes

### PR DESCRIPTION
**Pull request by zhani**
Also set data and cache paths explicitly so that they point to app private folder

Fixes [#126](https://github.com/Igalia/wpe-android/issues/126)

### CODE REVIEW

File: wpe/src/main/cpp/Browser/Page.cpp [hunk 1]
```diff
@@ -272,8 +278,11 @@ struct wpe_android_view_backend_exportable_client Page::s_exportableClient = {
         Browser::getInstance().handleExportedBuffer(*page, std::make_shared<ExportedBuffer>(buffer, poolID, bufferID));
     }};
 
-Page::Page(int width, int height, std::shared_ptr<PageEventObserver> observer)
-    : m_width(width)
+Page::Page(const std::string& dataDir, const std::string& cacheDir, int width, int height,
```
[the remark this answers is not in the document]
Comment (zhani) [reply wpe/src/main/cpp/Browser/Page.cpp]: Fixed

File: wpe/src/main/cpp/Browser/Page.cpp [hunk 2]
```diff
@@ -272,8 +278,11 @@ struct wpe_android_view_backend_exportable_client Page::s_exportableClient = {
         Browser::getInstance().handleExportedBuffer(*page, std::make_shared<ExportedBuffer>(buffer, poolID, bufferID));
     }};
 
-Page::Page(int width, int height, std::shared_ptr<PageEventObserver> observer)
-    : m_width(width)
+Page::Page(const std::string dataDir, const std::string cacheDir, int width, int height,
```
Comment (neodesys) [inline wpe/src/main/cpp/Browser/Page.cpp:281]: `std::string` and NOT `const std::string` just like I wrote in my previous comment

File: wpe/src/main/cpp/Browser/Page.cpp [hunk 3]
```diff
@@ -272,8 +278,11 @@ struct wpe_android_view_backend_exportable_client Page::s_exportableClient = {
...
+    std::shared_ptr<PageEventObserver> observer)
+    : m_dataDir(dataDir)
+    , m_cacheDir(cacheDir)
```
Comment (neodesys) [inline wpe/src/main/cpp/Browser/Page.cpp:284]: std::move()

File: wpe/src/main/cpp/Browser/Page.cpp [hunk 4]
```diff
@@ -272,8 +278,11 @@ struct wpe_android_view_backend_exportable_client Page::s_exportableClient = {
         Browser::getInstance().handleExportedBuffer(*page, std::make_shared<ExportedBuffer>(buffer, poolID, bufferID));
     }};
 
-Page::Page(int width, int height, std::shared_ptr<PageEventObserver> observer)
-    : m_width(width)
+Page::Page(const std::string dataDir, const std::string cacheDir, int width, int height,
+    std::shared_ptr<PageEventObserver> observer)
+    : m_dataDir(dataDir)
```
Comment (neodesys) [inline wpe/src/main/cpp/Browser/Page.cpp:283]: std::move()

**Review by neodesys — CHANGES_REQUESTED** [review]
Please re-read carefully what you are pushing :stuck_out_tongue_winking_eye:

**Review by zdobersek — APPROVED** [review]

**Review by neodesys — APPROVED** [review]

**Pull request merged**